src/lib.rs

```diff
@@ -98,17 +98,36 @@ pub const DEFAULT_MAX_ALLOWED_MSG_SIZE: usize = 500 * 1024 * 1024; // 500MiB
 /// before using a random port.
 pub const DEFAULT_PORT_TO_TRY: u16 = 443;
 
+/// Senders for node and client events
+#[derive(Clone)]
+pub struct EventSenders {
+    /// The client event sender
+    pub client_tx: mpmc::Sender<Event>,
+    /// The node event sender
+    pub node_tx: mpmc::Sender<Event>,
+}
+
+impl EventSenders {
+    pub(crate) fn send(&self, event: Event) -> Result<(), mpmc::SendError<Event>> {
```

Comment:
Adding this comment here but it applies generally to the whole design of this. There is an ergonomy issue in that the variants of the `Event` enum contain `Peer` which can be `Client` or `Node`, which means that even though you receive an event through say the `node` channel, you still need to at least nominally handle the `Client` case (using `unreachable!` for example). It would be better, API-wise, to have only `SocketAddr` instead of `Peer` in those `Event` variants and to use separate send functions: `send_client` and `send_node`.

Reply:
That would be fine with me, I think it is probably something that can be added by a follow up PR, as it is more of an additional change.

Reply:
Raised as https://github.com/maidsafe/routing/issues/2053